pallets/staking/src/lib.rs

```diff
@@ -1121,7 +1129,25 @@ pub enum Forcing {
 
 impl Default for Forcing {
     fn default() -> Self {
-        Forcing::NotForcing
+        Self::NotForcing
+    }
+}
+
+/// Switch that allows changing slashing status.
+#[derive(Copy, Clone, PartialEq, Eq, Encode, Decode, RuntimeDebug)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum SlashingSwitch {
+    /// Switch to change the slashing status for validator/operator.
+    Validator,
+    /// Switch to change the slashing status for Validator & Nominator both.
+    ValidatorAndNominator,
+    /// Neither the validator nor the nominator get slashed.
+    Off,
```

Comment:
`on_offence` only checks for the switch not to be `Off`. In fact, the code doesn't check directly for `Validator` anywhere apart from the unit tests. It may be more sensible to represent `Off` with a missing storage value and have only `Validator` and `ValidatorAndNominator` variants.

Reply:
I like to have an explicit variant this will help for the UI as well.